src/main/java/com/synopsys/integration/alert/component/settings/descriptor/SettingsUIConfig.java

```diff
@@ -127,4 +136,29 @@ public SettingsUIConfig(FilePersistenceUtil filePersistenceUtil) {
         }
         return List.of();
     }
+
+    private class EncryptionFieldsSetValidator extends EncryptionValidator {
+        @Override
+        public Collection<String> apply(final FieldValueModel fieldValueModel, final FieldModel fieldModel) {
+            Optional<FieldValueModel> pwdField = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_ENCRYPTION_PWD);
+            Optional<FieldValueModel> saltField = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_ENCRYPTION_GLOBAL_SALT);
+
+            boolean encryptionFieldsSet = pwdField.map(field -> field.hasValues() || field.isSet()).orElse(false)
+                                              && saltField.map(field -> field.hasValues() || field.isSet()).orElse(false);
```

Comment:
The optional fields here can be converted to booleans when initially stored then you can do a very basic check here instead of this busy one.

Reply:
You might also be able to use the !containsNoData here but I'm not sure if it's the same check or not